resources/js/core/browser-title-with-notification-count.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import { action, autorun, makeObservable, observable, runInAction } from 'mobx';
+import OsuCore from 'osu-core';
+import { formatNumber } from 'utils/html';
+
+export default class BrowserTitleWithNotificationCount {
+  private disposer?: () => void;
+  @observable private origTitle = '';
+
+  set title(newTitle: string) {
+    runInAction(() => {
+      this.origTitle = `${newTitle} | osu!`;
+    });
+  }
+
+  constructor(private core: OsuCore) {
+    document.addEventListener('turbolinks:load', this.setTitle);
+    document.addEventListener('turbolinks:before-cache', this.resetTitle);
+    makeObservable(this);
```

Comment:
`makeObservable` should be before the listeners get added otherwise the `@action` has no effect (which also does nothing by coincidence due to the `autorun` observing `origTitle` is always disposed when `setTitle` runs)